docs/Tutorials/Finding ROI average in Stack.md

```diff
@@ -0,0 +1,126 @@
+
+## Introduction
+
+ImageJS has the ability to decode a TIFF stack of images. A TIFF stack is a TIFF file that contains multiple images. In our specific case here, we have a stack of pulsar kind of images. They represent dynamic changes that happen to regions of interest frame-by-frame.
+We can use ImageJS to figure out when the region is visible and when it isn't by looking at the average value of said region compared to its maximum value in the stack.
+
+Let's go step by step together to figure out how it works.
+
+## Decode the Stack
+
+Just like any image, our stack needs to be decoded for us to work with data.
+
+```ts
+import * as fs from 'node:fs';
+
+const buffer = fs.readFileSync('/path/to/file.tiff');
+const stack = decodeStack(buffer);
+```
+
+:::warning
+`Stack` class works only with images that share same properties. Particularly, values for [bit depth](../Glossary.md#bit-depth 'internal link on bit depth'), [color model](../Glossary.md#color-model 'internal link on color model'), width and height must be the same.
+:::
+
+## Find the image with maximum values:
+
+Stack class has a function called `maxImage()`. It will give us the maximum value of each pixel throughout the stack. In our particular example we use `maxImage()`. The reason for it is that we want to look at the evaluation of pulsing particles. Therefore we need to detect all the appeared particles throughout the stack. We will use this image as a reference for all other images to locate their ROIs.
+
+```ts
+const maxValueImage = stack.maxImage();
+```
+
+![Image](./images/stackAvg/maxImage.png)
+
+## Locate ROIs
+
+From our `maxValueImage` we can find all regions of interest. To be precise we need their coordinates to apply them to other images.
+
+```ts
+//We will segment image through
+// `threshold()` to find ROIs,
+//therefore color model has to be "GREY".
+if (maxValueImage.colorModel !== 'GREY') {
+  maxValueImage = maxValueImage.grey();
+}
+const maxValueMask = maxValueImage.threshold();
+const roiMap = fromMask(maxValueMask);
+```
+
+![Mask](./images/stackAvg/maxMask.png)
+
+```ts
+//Provides all the regions of interest.
+const rois = roiMap.getRois();
+```
+
+![Painted ROIs](./images/stackAvg/paintedROIs.png)
+
+## Find average value of each ROI on each image
+
+After we found all the ROIs on the `maxValueImage`, we take each region and check its average intensity on every image in the stack.
+It will look something like this:
+
+```ts
+const stackGrays = new Map<number, number[][]>();
+for (const roi of rois) {
+  const stackAvgs = [];
+  const roiPoints = roi.absolutePoints;
+  for (const image of stack) {
+    const avgValue = image.mean({ points: roiPoints });
+    //Gets value from one channel since it is grayscaled.
+    stackAvgs.push(avgValue);
+  }
+  stackGrays.set(roi.id, stackAvgs);
+}
+```
+
+This will create a map where each key is an ROI ID, and each value is an array of average ROI intensities across the image stack.
```

Comment:
I think you are trying too hard to explain the code with technical details. You used a `Map` but it would have been fine to use an array for this as well.  Just explain what your are doing in generic terms. For example: "For each detected ROI, we compute the average value of the pixels in the ROI for each image in the stack. This gives us a time serie of intensities, which gives us an insight about when in time the detected objects appeared.